fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -283,7 +283,7 @@ class FullyShardedDataParallel(nn.Module):
     def __init__(
         self,
         module: nn.Module,
-        process_group: Optional[ProcessGroup] = None,
+        process_group: Optional[Dict[str, ProcessGroup]] = None,
```

Comment:
As we have discussed a bit last week, I think we can make this backward compatible. We can either (1) keep this unchanged and add a new param for the new PG or (2) make this argument check whether the type given is a PG or a Dict. I think perhaps (1) is easier and you basically add a new optional PG argument for reduce scatter. Also, please update the docstring and explain the reason for adding the new PG.

Reply:
(1) sounds good.